--- makefile.orig	2012-04-25 11:03:54.000000000 -0700
+++ makefile	2024-01-07 12:00:00.000000000 -0700
@@ -38,8 +38,8 @@
 ################
 ## TARGET OS HERE
 ################
-# OS Values: AIX, LINUX, SOLARIS, NCR, HPUX
-OS	=	LINUX
+# OS Values: AIX, LINUX, SOLARIS, NCR, HPUX, OSX
+OS	=	OSX
 ###########
 # No changes should be necessary below this point
 # Each compile variable is adjusted for the target platform using the OS setting above
@@ -47,12 +47,14 @@
 # CC
 AIX_CC		= xlC
 HPUX_CC		= gcc
-LINUX_CC		= gcc
+LINUX_CC	= gcc
+OSX_CC		= gcc
 NCR_CC		= cc
 SOLARIS_CC	= gcc
 SOL86_CC	= cc
 CC		= $($(OS)_CC)
 # CFLAGS
+OSX_CFLAGS	= -g -Wall -I./include
 AIX_CFLAGS		= -q64 -O3 -D_LARGE_FILES
 HPUX_CFLAGS		= -O3 -Wall
 LINUX_CFLAGS	= -g -Wall
@@ -65,6 +67,7 @@
 AIX_EXE	=
 HPUX_EXE	=
 LINUX_EXE	=
+OSX_EXE	=
 NCR_EXE		=
 SOLARIS_EXE	=
 SOL86_EXE	=
@@ -73,6 +76,7 @@
 AIX_LEX		= flex
 HPUX_LEX	= flex
 LINUX_LEX	= lex
+OSX_LEX		= lex
 NCR_LEX		= lex
 SOLARIS_LEX	= lex
 SOL86_LEX	= lex
@@ -81,6 +85,7 @@
 AIX_LIBS	= -lm
 HPUX_LIBS	= -lm -ll
 LINUX_LIBS	= -lm
+OSX_LIBS	= -lm
 NCR_LIBS	= -lm -lc89
 SOLARIS_LIBS	= -ly -ll -lm
 SOL86_LIBS	= -ly -ll -lm
@@ -89,6 +94,7 @@
 AIX_YACC	= yacc
 HPUX_YACC	= bison -y
 LINUX_YACC	= yacc
+OSX_YACC	= yacc
 NCR_YACC	= yacc
 SOLARIS_YACC	= yacc
 SOL86_YACC	= yacc
@@ -97,6 +103,7 @@
 AIX_YFLAGS	= -d -v
 HPUX_YFLAGS	= -y -d -v
 LINUX_YFLAGS	= -d -v
+OSX_YFLAGS	= -d -v
 NCR_YFLAGS	= -d -v
 SOLARIS_YFLAGS	= -d -v
 SOL86_YFLAGS	= -d -v
--- Makefile.suite.orig	2012-04-25 11:03:50.000000000 -0700
+++ Makefile.suite	2024-01-07 12:00:00.000000000 -0700
@@ -38,8 +38,8 @@
 ################
 ## TARGET OS HERE
 ################
-# OS Values: AIX, LINUX, SOLARIS, NCR, HPUX
-OS	=	LINUX
+# OS Values: AIX, LINUX, SOLARIS, NCR, HPUX, OSX
+OS	=	OSX
 ###########
 # No changes should be necessary below this point
 # Each compile variable is adjusted for the target platform using the OS setting above
@@ -47,12 +47,14 @@
 # CC
 AIX_CC		= xlC
 HPUX_CC		= gcc
-LINUX_CC		= gcc
+LINUX_CC	= gcc
+OSX_CC		= gcc
 NCR_CC		= cc
 SOLARIS_CC	= gcc
 SOL86_CC	= cc
 CC		= $($(OS)_CC)
 # CFLAGS
+OSX_CFLAGS	= -g -Wall -I./include
 AIX_CFLAGS		= -q64 -O3 -D_LARGE_FILES
 HPUX_CFLAGS		= -O3 -Wall
 LINUX_CFLAGS	= -g -Wall
@@ -65,6 +67,7 @@
 AIX_EXE	=
 HPUX_EXE	=
 LINUX_EXE	=
+OSX_EXE	=
 NCR_EXE		=
 SOLARIS_EXE	=
 SOL86_EXE	=
@@ -73,6 +76,7 @@
 AIX_LEX		= flex
 HPUX_LEX	= flex
 LINUX_LEX	= lex
+OSX_LEX		= lex
 NCR_LEX		= lex
 SOLARIS_LEX	= lex
 SOL86_LEX	= lex
@@ -81,6 +85,7 @@
 AIX_LIBS	= -lm
 HPUX_LIBS	= -lm -ll
 LINUX_LIBS	= -lm
+OSX_LIBS	= -lm
 NCR_LIBS	= -lm -lc89
 SOLARIS_LIBS	= -ly -ll -lm
 SOL86_LIBS	= -ly -ll -lm
@@ -89,6 +94,7 @@
 AIX_YACC	= yacc
 HPUX_YACC	= bison -y
 LINUX_YACC	= yacc
+OSX_YACC	= yacc
 NCR_YACC	= yacc
 SOLARIS_YACC	= yacc
 SOL86_YACC	= yacc
@@ -97,6 +103,7 @@
 AIX_YFLAGS	= -d -v
 HPUX_YFLAGS	= -y -d -v
 LINUX_YFLAGS	= -d -v
+OSX_YFLAGS	= -d -v
 NCR_YFLAGS	= -d -v
 SOLARIS_YFLAGS	= -d -v
 SOL86_YFLAGS	= -d -v
--- config.h.orig	2012-04-25 11:03:52.000000000 -0700
+++ config.h	2024-01-07 12:00:00.000000000 -0700
@@ -109,6 +109,18 @@
 #define FLEX
 #endif /* LINUX */

+#ifdef OSX
+#define SUPPORT_64BITS
+#define HUGE_TYPE       int64_t
+#define HUGE_FORMAT     "%lld"
+#define HUGE_COUNT      1
+#define USE_STRING_H
+#define USE_LIMITS_H
+#define MAXINT INT_MAX
+#define USE_STDLIB_H
+#define FLEX
+#endif /* OSX */
+
 #ifdef SOLARIS
 #define SUPPORT_64BITS
 #define HUGE_TYPE	long long
--- date.c.orig	2012-04-25 11:03:52.000000000 -0700
+++ date.c	2024-01-07 12:00:00.000000000 -0700
@@ -563,7 +563,7 @@
  * TODO: None
  */
 int
-getDateWeightFromJulian(jDay, nDistribution)
+getDateWeightFromJulian(int jDay, int nDistribution)
 {
 	date_t dTemp;
 	int nDay;
--- tdef_functions.c.orig	2012-04-25 11:03:52.000000000 -0700
+++ tdef_functions.c	2024-01-07 12:00:00.000000000 -0700
@@ -171,7 +171,7 @@
 };

 table_func_t *
-getTdefFunctionsByNumber(nTable)
+getTdefFunctionsByNumber(int nTable)
 {
    if (nTable >= S_BRAND)
       return(&s_tdef_funcs[nTable - S_BRAND]);
--- tdefs.c.orig	2012-04-25 11:03:52.000000000 -0700
+++ tdefs.c	2024-01-07 12:00:00.000000000 -0700
@@ -147,7 +147,7 @@
 }
 */
 tdef *
-getSimpleTdefsByNumber(nTable)
+getSimpleTdefsByNumber(int nTable)
 {
    if (nTable >= S_BRAND)
       return(&s_tdefs[nTable - S_BRAND]);
